Xplat/src/main/java/vazkii/botania/common/item/material/EnderAirItem.java

```diff
@@ -43,22 +49,32 @@ public static InteractionResultHolder<ItemStack> onPlayerInteract(Player player,
 			return InteractionResultHolder.pass(stack);
 		}
 
-		if ((world.dimension() == Level.END && isClearFromDragonBreath(world, player.getBoundingBox().inflate(3.5)))
+		if ((world.dimension() == Level.END && isClearFromDragonBreath(world, player.getBoundingBox().inflate(3.5)) && notAimingAtFluid(world, player))
 				|| pickupFromEntity(world, player.getBoundingBox().inflate(1.0))) {
 
 			if (!world.isClientSide) {
 				ItemStack enderAir = new ItemStack(BotaniaItems.enderAirBottle);
 				player.getInventory().placeItemBackInInventory(enderAir);
 				stack.shrink(1);
 				world.playSound(null, player.blockPosition(), SoundEvents.ITEM_PICKUP, SoundSource.NEUTRAL, 0.5F, 1F);
+				world.gameEvent(player, GameEvent.FLUID_PICKUP, player.position());
 			}
 
-			return InteractionResultHolder.success(stack);
+			return InteractionResultHolder.sidedSuccess(stack, world.isClientSide());
 		}
 
 		return InteractionResultHolder.pass(stack);
 	}
 
+	private static boolean notAimingAtFluid(Level world, Player player) {
+		BlockHitResult hitResult = getPlayerPOVHitResult(world, player, ClipContext.Fluid.SOURCE_ONLY);
+		if (hitResult.getType() == HitResult.Type.BLOCK) {
+			BlockPos pos = hitResult.getBlockPos();
+			return world.mayInteract(player, pos) && !world.getFluidState(pos).is(FluidTags.WATER);
```

Comment:
why is this mayInteract here? Pretty sure mc internals already do this check before calling any item specific stuff.

Reply:
also, is this meant to only check for water or absence of any fluid? the logic doesn't match the method name